src/wp-includes/template-loader.php

```diff
@@ -1,4 +1,44 @@
 <?php
+
+ob_start(
+	static function ( string $output, ?int $phase ): string {
+		// When the output is being cleaned (e.g. pending template is replaced with error page), do not send it through the filter.
+		if ( ( $phase & PHP_OUTPUT_HANDLER_CLEAN ) !== 0 ) {
+			return $output;
+		}
+
+		/**
+		 * Filters the template output buffer prior to sending to the client.
+		 *
+		 * The output buffer is started before the `template_redirect` action is triggered, allowing templates rendered
+		 * at that action to also have their output filtered.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param string $output Output buffer.
+		 * @return string Filtered output buffer.
+		 */
+		return (string) apply_filters( 'wp_template_output_buffer', $output );
```

Comment:
Actually, for caching plugins they shouldn't rely on the filter to capture the output since there could be another plugin that tries filtering after they do the capturing. Instead there should be an action that fires after the filters apply for caching plugins to be told what the output is:

```suggestion
		 */
		$output = (string) apply_filters( 'wp_template_output_buffer', $output );
		
		/**
		 * Fires after the output buffer has been filtered prior to sending to the client.
		 *
		 * This is useful for caching plugins to capture the page output for storage.
		 *
		 * @since n.e.x.t
		 *
		 * @param string $output Output buffer.
		 */
		 do_action( 'wp_final_template_output_buffer', $output );
		 return $output;
```